### PR TITLE
Onboarding

### PR DESCRIPTION
**Pull request by ksheremet**
screenshot_20181015-184608
screenshot_20181015-184612
screenshot_20181015-184617

### CODE REVIEW

File: flutter/lib/views/onboarding/intro_view.dart [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  void initState() {
+    super.initState();
+    _prefs.then((pref) {
+      _isIntroShown = pref.getBool(_introPrefKey);
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    if (_isIntroShown != null && _isIntroShown != false) {
```
Comment (dotdoom) [inline flutter/lib/views/onboarding/intro_view.dart]: `if (_isIntroShown == true)` ?

File: flutter/lib/views/onboarding/intro_view.dart [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+class _OnboardingViewWidgetState extends State<OnboardingViewWidget> {
+  static const String _introPrefKey = 'is-intro-shown';
+  final Future<SharedPreferences> _prefs = SharedPreferences.getInstance();
+  bool _isIntroShown;
+
+  @override
+  void initState() {
+    super.initState();
+    _prefs.then((pref) {
```
Comment (dotdoom) [inline flutter/lib/views/onboarding/intro_view.dart]: `_isIntroShown = (await _prefs).getBool(_introPrefKey)`

File: flutter/lib/views/onboarding/intro_view.dart [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    if (_isIntroShown != null && _isIntroShown != false) {
+      return SignInWidget(child: widget.child);
+    } else {
+      return IntroViewWidget(callback: () {
+        setState(() {
+          _prefs.then((pref) {
```
Comment (dotdoom) [inline flutter/lib/views/onboarding/intro_view.dart]: same (`await _prefs`)

File: flutter/pubspec.yaml [hunk 4]
```diff
@@ -26,6 +27,9 @@ dependencies:
   firebase_messaging: 2.0.1
   google_sign_in: 3.2.1
 
+  # Onboarding
+  intro_views_flutter: ^2.2.4
```
Comment (dotdoom) [inline flutter/pubspec.yaml]: Move to where the rest of dependencies are?

File: flutter/lib/main.dart [hunk 5]
```diff
@@ -34,10 +34,11 @@ class App extends StatelessWidget {
         Locale('en', 'US'),
         Locale('ru', 'RU'),
       ],
-      navigatorObservers: [_analyticsNavigatorObserver],
+      navigatorObservers: [App._analyticsNavigatorObserver],
```
Comment (dotdoom) [inline flutter/lib/main.dart]: This change is not necessary anymore?

File: flutter/lib/views/onboarding/intro_view.dart [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+            pref.setBool(_introPrefKey, true);
+            _isIntroShown = true;
+          });
+        });
+      });
+    }
+  }
+}
+
+class IntroViewWidget extends StatelessWidget {
```
Comment (dotdoom) [inline flutter/lib/views/onboarding/intro_view.dart]: Make this a private class (`_IntroViewWidget`)?

File: flutter/lib/views/onboarding/intro_view.dart [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+  static const double _imageHeight = 285.0;
```
Comment (dotdoom) [inline flutter/lib/views/onboarding/intro_view.dart]: Wouldn't it be so much better to specify width [in percent](https://stackoverflow.com/questions/43122113/sizing-elements-to-percentage-of-screen-width-height)?

**Comment by ksheremet** [comment]
With Russian locale

screenshot_20181119-210216
screenshot_20181119-210220
screenshot_20181119-210234

**Pull request merged**